### PR TITLE
Deterministic order verifiers/changed_keys

### LINKED ISSUE #891: ledger/protocol: sort changed storage keys
the changed storage keys `HashSet`s that are being passed to VPs (`keys_changed: HashSet<storage::Key>`) must be sorted to avoid non-deterministic VP results (we can use the derived `Ord` on `storage::Key` for sorting)

### PR DESCRIPTION
**Pull request by Fraccaman**
Switching from `Hashset` to `BTreeSet` for `verifiers` and `changed_keys`. Fix #891.

### CODE REVIEW

**Comment by Fraccaman** [comment]
pls update wasm

**Comment by Fraccaman** [comment]
pls update wasm

**Comment by tzemanovic** [comment]
if you could pls also update `docs/src/explore/design/ledger/vp.md:21` and `docs/src/user-guide/ledger/customize.md:46`

**Review by tzemanovic — APPROVED** [review]
LGTM, thanks! (still to add changelog + docs updates)

File: .changelog/unreleased/features/891-vp-order-deterministic.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- Fix changed_keys and verifiers order deterministic
```
Comment (tzemanovic) [inline .changelog/unreleased/features/891-vp-order-deterministic.md:1]:
```suggestion
- Change the validity predicate main entry-point function to receive `changed_keys` and `verifiers` arguments in a deterministic order.
```
it should also go under `bugs`, not `features`

Comment (tzemanovic) [inline .changelog/unreleased/features/891-vp-order-deterministic.md:1]:
```suggestion
- Change the validity predicate main entry-point function to receive `changed_keys` and `verifiers` arguments in a deterministic order.
```
it should also go under `bugs`, not `features`

**Pull request merged**